packages/hardhat-core/src/types/artifacts.ts

```diff
@@ -97,6 +97,27 @@ export interface Artifacts {
    * @param fullyQualifiedName The FQN of the artifact.
    */
   formArtifactPathFromFullyQualifiedName(fullyQualifiedName: string): string;
+
+  /**
+   * Starting with Hardhat 2.11.0, the artifacts object caches the information
+   * about paths that it fetches from the filesystem (e.g. the list of
+   * artifacts, the path that an artifact name resolves to, etc.). The artifacts
+   * and buildInfos themselves are not cached, only their paths.
+   *
+   * This method, if present, clears that cache.
+   */
+  clearCache?: () => void;
```

Comment:
Should we be calling this in the `clean` task?

Reply:
Ohhh! I hadn't thought about it. Yes!